src/ppx_deriving_yojson.cppo.ml

```diff
@@ -406,6 +414,54 @@ let desu_type_of_decl ~options ~path type_decl =
                        (fun var -> [%type: Yojson.Safe.json -> [%t error_or var]]) type_decl in
   polymorphize [%type: Yojson.Safe.json -> [%t error_or typ]]
 
+let desu_str_of_record ~is_strict ~error ~path wrap_record labels =
+  let top_error = error path in
+  let record =
+    List.fold_left
+      (fun expr i ->
+        [%expr [%e evar (argn i)] >>= fun [%p pvar (argn i)] -> [%e expr]]
+      )
+      [%expr
```

Comment:
The expression starting at this line is somewhat unwieldy, can you make it more readable?